src/state/routing/v2Slice.ts

```diff
@@ -0,0 +1,120 @@
+import { createApi, fetchBaseQuery, FetchBaseQueryError } from '@reduxjs/toolkit/query/react'
+import { Protocol } from '@uniswap/router-sdk'
+import { getClientSideQuote } from 'lib/hooks/routing/clientSideSmartOrderRouter'
+import ms from 'ms.macro'
+import { trace } from 'tracing/trace'
+
+import { GetQuoteArgs, INTERNAL_ROUTER_PREFERENCE_PRICE, RouterPreference } from './slice'
+import { QuoteDataV2, QuoteState, TradeResult } from './types'
+import { getRouter, isExactInput, shouldUseAPIRouter, transformRoutesToTrade } from './utils'
+
+const CLIENT_PARAMS = {
+  protocols: [Protocol.V2, Protocol.V3, Protocol.MIXED],
+}
+
+// routing API quote query params: https://github.com/Uniswap/routing-api/blob/main/lib/handlers/quote/schema/quote-schema.ts
+const CLASSIC_SWAP_QUERY_PARAMS = {
+  ...CLIENT_PARAMS,
+  routingType: 'CLASSIC',
+}
+
+export const routingApiV2 = createApi({
+  reducerPath: 'routingApiV2',
+  baseQuery: fetchBaseQuery({
+    baseUrl: 'https://api.uniswap.org/v2/',
+  }),
+  endpoints: (build) => ({
+    getQuote: build.query<TradeResult, GetQuoteArgs>({
+      async onQueryStarted(args: GetQuoteArgs, { queryFulfilled }) {
+        trace(
+          'quote-v2',
+          async ({ setTraceError, setTraceStatus }) => {
+            try {
+              await queryFulfilled
+            } catch (error: unknown) {
+              if (error && typeof error === 'object' && 'error' in error) {
+                const queryError = (error as Record<'error', FetchBaseQueryError>).error
+                if (typeof queryError.status === 'number') {
+                  setTraceStatus(queryError.status)
+                }
+                setTraceError(queryError)
+              } else {
+                throw error
+              }
+            }
+          },
+          {
+            data: {
+              ...args,
+              isPrice: args.routerPreference === INTERNAL_ROUTER_PREFERENCE_PRICE,
+              isAutoRouter:
```

Comment:
nit but seems same as `shouldUseAPIRouter`

Reply:
🤔 yeah not sure what this data tag is for, i think in the future we could just track the routerPreference instead of these boolean tags